awx/main/notifications/webhook_backend.py

```diff
@@ -32,14 +38,19 @@ def send_messages(self, messages):
         sent_messages = 0
         if 'User-Agent' not in self.headers:
             self.headers['User-Agent'] = "Tower {}".format(get_awx_version())
+        if self.http_method.lower() not in ('put', 'post'):
+            raise ValueError("HTTP method must be either 'POST' or 'PUT'.")
+        chosen_method = getattr(requests, self.http_method.lower(), None)
         for m in messages:
-            r = requests.post("{}".format(m.recipients()[0]),
-                              json=m.body,
-                              headers=self.headers,
-                              verify=(not self.disable_ssl_verification))
-            if r.status_code >= 400:
-                logger.error(smart_text(_("Error sending notification webhook: {}").format(r.text)))
-                if not self.fail_silently:
-                    raise Exception(smart_text(_("Error sending notification webhook: {}").format(r.text)))
-            sent_messages += 1
+            if chosen_method is not None:
+                r = chosen_method("{}".format(m.recipients()[0]),
+                                  auth=(self.username, self.password),
```

Comment:
We should change this to only happen **iff** these have values i.e.,

```python
auth = None
if self.username and self.password:
    auth = (self.username, self.password)
```

When these values are empty/missing, we should *not* send a header.

Reply:
Webhooks are no longer sending an Authorization header if the username/password fields are blank!